js/customize-snapshots.js

```diff
@@ -791,6 +793,141 @@
 		return date.valueOf();
 	};
 
+	component.conflict = {
+		warningTemplate: wp.template( 'snapshot-conflict' ),
+		icon: wp.template( 'snapshot-conflict-button' ),
+		refreshBuffer: 250,
+		controls: {},
+		pendingRequest: {}
+	};
+
+	/**
+	 * Handles snapshot conflict events.
+	 *
+	 * @return {void}
+	 */
+	component.handleEventForConflicts = function() {
+		api.state( 'saved' ).bind(function( saved ) {
+			if ( saved ) {
+				_.each( component.conflict.controls, function( control ) {
+					control.remove();
+				} );
+				$( '.snapshot-conflicts-button' ).hide();
+				if ( component.conflict._currentRequest ) {
+					component.conflict._currentRequest.abort();
+					component.conflict._currentRequest = null;
+				}
+				component.conflict.controls = {};
+				component.conflict.pendingRequest = {};
+
+				wp.customize.control.each( function( control ) {
+					_.each( control.settings, function( setting ) {
+						setting.unbind( component.onConflictFirstChange );
+					} );
+					component.addConflictButton( control );
+				} );
+			}
+		});
+	};
+
+	/**
+	 * Add conflict button icon on first change and trigger handleConflictRequest.
+	 *
+	 * @param {object} control where conflict button will be added.
+	 *
+	 * @return {void}
+	 */
+	component.addConflictButton = function addConflictButton( control ) {
+		control.deferred.embedded.done( function() {
+			var onFirstChange, hasDirty, buttonTemplate;
+
+			if ( ! control.setting ) {
+				return;
+			}
+
+			onFirstChange = component.onConflictFirstChange = function() {
+				_.each( control.settings, function( setting ) {
+					setting.unbind( onFirstChange );
+					if ( ! control.container.find( '.snapshot-conflicts-button' ).length ) {
+						buttonTemplate = $( $.trim( component.conflict.icon( {
+							setting_id: setting.id
+						} ) ) );
+						buttonTemplate.hide();
+						control.container.find( '.customize-control-title' ).append( buttonTemplate );
```

Comment:
There is a danger here for adding a button to the title because a control may actually not have a title. However, all controls will have a… notification area. I just realized an alternative to adding a custom button like this it could be better to make use of the notification area. A notification of type `warning` could be added as depicted here: https://github.com/xwp/wp-customize-posts/issues/217#issuecomment-239746179